web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -448,6 +448,50 @@ const EncryptionSettingsForm = ({
   );
 };
 
+/**
+ * Allows to define snapshots enablement
+ * @component
+ *
+ * @param {object} props
+ * @param {boolean} [props.isChecked=false] - Whether system snapshots are selected
+ * @param {boolean} [props.isLoading=false] - Whether to show the selector as loading
+ * @param {onChangeFn} [props.onChange=noop] - On change callback
+ *
+ * @callback onChangeFn
+ * @param {object} settings
+ */
+const SnapshotsField = ({
+  isChecked: isCheckedProp = false,
+  isLoading = false,
+  onChange = noop
+}) => {
+  const [isChecked, setIsChecked] = useState(isCheckedProp);
+
+  const switchState = (checked) => {
+    setIsChecked(checked);
+    onChange(checked);
+  };
+
+  if (isLoading) return <Skeleton width="25%" />;
+
+  const explanation = _("Btrfs snapshots for root partition enforce root volume to use btrfs with snapshots enabled. It is useful for returning to previous snapshot of system e.g. after software upgrade.");
```

Comment:
Broken English, missing a lot of articles. This should be:

"Btrfs snapshots for **the** root partition enforce **the** root volume to use btrfs with snapshots enabled. ~~It~~ This is useful for returning to **a** previous snapshot of **the** system **,** e.g. after **a** software upgrade."

Maybe also add something about problems created by the admin to point out that it's not usually problems that our upgrades introduced:

"...or administrator actions that turned out to be problematic."

or

"...or manual configurations that turned out to be problematic."